test/cli/transports_test.go

```diff
@@ -0,0 +1,127 @@
+package cli
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ipfs/kubo/config"
+	"github.com/ipfs/kubo/test/cli/harness"
+	"github.com/ipfs/kubo/test/cli/testutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTransports(t *testing.T) {
+	disableRouting := func(nodes harness.Nodes) {
+		nodes.ForEachPar(func(n *harness.Node) {
+			n.UpdateConfig(func(cfg *config.Config) {
+				cfg.Routing.Type = config.NewOptionalString("none")
+				cfg.Bootstrap = nil
+			})
+		})
+	}
+	checkSingleFile := func(nodes harness.Nodes) {
+		s := testutils.RandomStr(100)
+		hash := nodes[0].IPFSAddStr(s)
+		nodes.ForEachPar(func(n *harness.Node) {
+			val := n.IPFS("cat", hash).Stdout.Trimmed()
+			assert.Equal(t, s, val)
+		})
+	}
+	checkRandomDir := func(nodes harness.Nodes) {
+		randDir := filepath.Join(nodes[0].Dir, "foobar")
+		require.NoError(t, os.Mkdir(randDir, 0777))
+		rf := testutils.NewRandFiles()
+		rf.FanoutDirs = 3
+		rf.FanoutFiles = 6
+		require.NoError(t, rf.WriteRandomFiles(randDir, 4))
+
+		hash := nodes[1].IPFS("add", "-r", "-Q", randDir).Stdout.Trimmed()
```

Comment:
`.Trimmed` too here.

Reply:
I think `Trimmed` is correct to use here, as the CLI writes a newline char to stdout that needs to be removed. This is not the case for `ipfs cat`. The sharness tests do the same thing, except the trimming happens in bash, which strips the trailing newline as part of command substitution.